XenServerHealthCheck/ServerListHelper.cs

```diff
@@ -165,13 +165,28 @@ public bool UpdateServerInfo(ServerInfo server)
             }
         }
 
+        public void ClearServerList()
+        {
+            lock (serverListLock)
+            {
+                serverList.Clear();
+                updateServerList();
+            }
+        }
+
         public void UpdateServerCredential(string credential)
         {
             log.Info("Receive credential update message");
 
             string decryptCredential = EncryptionUtils.UnprotectForLocalMachine(credential);
             string[] decryptCredentialComps = decryptCredential.Split(SEPARATOR);
 
+            if (decryptCredentialComps.Length != 1 && decryptCredentialComps.Length != 3)

```

Comment:
Would this a good practice to hardcode the length inside? I am not sure, just wondering.